src/App.js

```diff
@@ -30,36 +30,84 @@ const App = () => {
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const [isX, setIsX] = useState(false);
+
+  const currentPlayer = !isX ? player1 : player2;
+
+  const updateSquare = (id) => {
+    let newSquares = [];
+
+    for (let row = 0; row < 3; row += 1) {
+      newSquares.push([]);
+      for (let col = 0; col < 3; col += 1) {
+        if (id === squares[row][col].id) {
+          if (!squares[row][col].value) {
+            squares[row][col].value = currentPlayer;
+            setIsX(!isX);
+          }
+        }
+        newSquares[row].push(squares[row][col]);
+      }
+    }
+    setSquares(newSquares);
+  };
```

Comment:
Great job in making a copy of the squares data! Here's a reminder on why we need to make the copy:

To trigger the `update` stage for this component, React needs to notice a change in state data. We could mutate state directly, however, this doesn't scale well and doesn't follow React best practices. 

Instead, we should pass in a copy of our state data to the setState function `setSquares()` which will have React compare the object references between the copy and the old state data. If the object references are different, React will then trigger the `update` stage.

Reply:
Rather than creating an empty list and pushing values into it with every iteration, we can make a shallow copy with the `...` spread operator:

```suggestion
    let newSquares = [...squares];

    for (let row = 0; row < 3; row += 1) {
      for (let col = 0; col < 3; col += 1) {
        if (id === squares[row][col].id && !squares[row][col].value) {
            squares[row][col].value = currentPlayer;
            setIsX(!isX);
        }
      }
    }
    setSquares(newSquares);
  };
```

Reply:
I've also combined the conditional to reduce how _nested_ this section of code looks.